File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/utils/CredentialsValidator.java
package hr.fer.oobl.iorder.iorder.utils;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.text.TextUtils;

public class CredentialsValidator {

    public static boolean validateEmail(String email) {
        return !TextUtils.isEmpty(email.trim())
                && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }

    public static boolean validatePassword(String password) {
        return !TextUtils.isEmpty(password) && password.length() >= 8;
    }

    public static boolean checkInternetConnection(Activity activity) {
        ConnectivityManager connectivityManager =
                (ConnectivityManager) activity.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
        if (networkInfo != null && networkInfo.isConnected()) {
            return true;
        }

        return false;
    }

    public static boolean validateNickname(String nickname) {
        return !TextUtils.isEmpty(nickname.trim());
    }

    public static boolean comparePasswords(String password, String confPass) {
        return password.equals(confPass);
    }
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/utils/CredentialsValidator.kt
package hr.fer.oobl.iorder.iorder.utils

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.text.TextUtils

object CredentialsValidator {

    fun validateEmail(email: String): Boolean {
        return !TextUtils.isEmpty(email.trim { it <= ' ' }) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }

    fun validatePassword(password: String): Boolean {
        return !TextUtils.isEmpty(password) && password.length >= 8
    }

    fun checkInternetConnection(activity: Activity): Boolean {
        val connectivityManager = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        return networkInfo != null && networkInfo.isConnected

    }

    fun validateNickname(nickname: String): Boolean {
        return !TextUtils.isEmpty(nickname.trim { it <= ' ' })
    }

    fun comparePasswords(password: String, confPass: String): Boolean {
        return password == confPass
    }
}
